scripts/generate.sh

```diff
@@ -14,11 +14,24 @@ done
 
 # jar was downloaded from here https://repo1.maven.org/maven2/io/swagger/codegen/v3/swagger-codegen-cli/3.0.34/
 
-if [[ $VERSION == *"develop"* ]]; then
-  SANITIZED_VERSION=$(sed 's/-develop//g' <<< $VERSION)
-  SANITIZED_VERSION=$(sed 's/\.[0-9]*$//g' <<< $SANITIZED_VERSION)
-  BUILD_VERSION=$(grep -o '[0-9]*$' <<< $VERSION)
-  VERSION="dev-$SANITIZED_VERSION-$BUILD_VERSION"
+VERSION=${VERSION//develop/dev}
+
+# Example for the regex below:
+# dev.1.0.0.0
+# dev-1.0.0-0
+# dev.1.0.0-0
+# dev-1.0.0.0
+if [[ $VERSION =~ ^dev[.-]([0-9]+)[.-]([0-9]+)[.-]([0-9]+)[.-]([0-9]+)$ ]]; then
+    VERSION="${BASH_REMATCH[1]}.${BASH_REMATCH[2]}.${BASH_REMATCH[3]}-beta.${BASH_REMATCH[4]}"
+elif [[ $VERSION =~ ^([0-9]+)[.-]([0-9]+)[.-]([0-9]+)[.-]dev[.-]([0-9]+)$ ]]; then
```

Comment:
I expect here and for the following regexps examples too =)

Reply:
Updated